### PR TITLE
kvm: Passthrough host CPU in order to allow nesting

### LINKED ISSUE #2553: Enable KVM nesting
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Unsure

Please provide the following details:

**Environment**: Fedora 26

**Minikube version**: v0.25.0
- **OS** (e.g. from /etc/os-release):
- **VM Driver**: kvm2
- **ISO version**: minikube-v0.25.1.iso
- **Install tools**:
- **Others**:

**What happened**: need /dev/kvm inside minikube

**What you expected to happen**: /dev/kvm is not inside

**How to reproduce it** (as minimally and precisely as possible): launch minikube look for /dev/kvm (have netsing on the host enabled

**Output of `minikube logs` (if applicable)**:

**Anything else do we need to know**: According to docker-machine-kvm#36 nesting should be at least supported by docker machine.

### PR DESCRIPTION
**Pull request by fabiand**
Before this patch the virtual CPU was the stock qemu CPU, in the sense that
the virtual CPU features were set according to the default qemu CPU.

With this change the CPU features of the host will be copied at start to the
domain definition. This includes features like svm and vmx, which in turn allow
to run nested virtualization if the host is configured accordingly i.e.
`kvm_intel nested=y` in `/etc/modprobe.d/kvm.conf`.

To turn on nesting, a user has to specify `--kvm-cpu-model host-model` when
creating the VM.

Resolves #2553

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

### CODE REVIEW

**Comment by fabiand** [comment]
Should be in a reviewable state now.

**Comment by fabiand** [comment]
I cna confirm that nesting works:

```
$ minikube ssh
                         _             _            
            _         _ ( )           ( )           
  ___ ___  (_)  ___  (_)| |/')  _   _ | |_      __  
/' _ ` _ `\| |/' _ `\| || , <  ( ) ( )| '_`\  /'__`\
| ( ) ( ) || || ( ) || || |\`\ | (_) || |_) )(  ___/
(_) (_) (_)(_)(_) (_)(_)(_) (_)`\___/'(_,__/'`\____)

$ ls -shal /dev/kvm 
0 crw-rw---- 1 root root 10, 232 Feb 16 18:54 /dev/kvm
```

Achieved with this patchset as is.

**Comment by r2d4** [comment]
@minikube-bot ok to test

**Comment by fabiand** [comment]
Not sure why the tests fail.

**Comment by fabiand** [comment]
@zakame :)

I chose `custom` as it is compatible with the current settings.
We can choose to go with `host-model` which should also be safe (but different to the current settings), and lead to much better nesting performance.

Even if I favor to have `host-model` or `host-passthrough` by default, I tend to say that we can default to `custom` for a release or two, and then eventually do th switch if there was no negative feedback from `host-*` users.

File: pkg/minikube/cluster/cluster_linux.go [hunk 1]
```diff
@@ -71,6 +72,7 @@ func createKVM2Host(config MachineConfig) *kvmDriver {
 		},
 		Memory:         config.Memory,
 		CPU:            config.CPUs,
+		CPUModel:       config.KvmCPUModel,
```
Comment (dlorenc) [inline pkg/minikube/cluster/cluster_linux.go]: Is there any reason to actually pass this through as a config option? Could we hardcode it in the driver XML?

Comment (fabiand) [reply pkg/minikube/cluster/cluster_linux.go]: We could hard code it if we are willing to make this change.

@michalskrivanek @berrange do you actually see a risk of defaulting to `host-model`? (Could you also confirm that host-model will enable nesting [assuming the host is configured accordingly])

Comment (berrange) [reply pkg/minikube/cluster/cluster_linux.go]: Since you dont care about migration you should use  host-passthrough in preference to host-model.

**Comment by fabiand** [comment]
@dlorenc @zakame @berrange I updated the change set to set `host-passthrough` unconditionally/by default.

**Comment by fabiand** [comment]
Any opinion on this PR?

I'd be fine in going with teh hard coded solution, there should be no drawback.

**Comment by fabiand** [comment]
Awesome!

**Pull request merged**